pkg/identifiers/validate.go

```diff
@@ -42,9 +42,5 @@ func ValidateDockerCompat(s string) error {
 		return fmt.Errorf("identifier %q must match pattern %q: %w", s, AllowedIdentfierChars, errdefs.ErrInvalidArgument)
 	}
 
-	if err := validatePlatformSpecific(s); err != nil {
```

Comment:
Platform validation of identifiers is a filesystem concern.
These are moved into the `store.Store`, as they are orthogonal concerns.